File: app/src/main/java/org/worshipsongs/service/DatabaseService.java
package org.worshipsongs.service;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import org.apache.commons.lang3.StringUtils;
import org.worshipsongs.helper.DatabaseHelper;
import org.worshipsongs.utils.RegexUtils;

import java.io.IOException;

/**
 * @Author : Madasamy
 * @Version : 1.0
 */

public class DatabaseService
{
    private SQLiteDatabase database;
    private DatabaseHelper databaseHelper;
    public static final String TOPIC_NAME_REGEX = "\\{.*\\}";

    public DatabaseService()
    {
        //Do nothing
    }

    public DatabaseService(Context context)
    {
        databaseHelper = new DatabaseHelper(context);
    }

    public void copyDatabase(String databasePath, boolean dropDatabase) throws IOException
    {
        databaseHelper.createDataBase(databasePath, dropDatabase);
    }

    public boolean isDatabaseExist()
    {
        return databaseHelper.checkDataBase();
    }

    public void open()
    {
        database = databaseHelper.openDataBase();
    }

    public void close()
    {
        databaseHelper.close();
    }

    public SQLiteDatabase getDatabase()
    {
        if (database == null) {
            database = databaseHelper.openDataBase();
        }
        return database;
    }

   public String parseTamilName(String topicName)
    {
        if (StringUtils.isNotBlank(topicName)) {
            String tamilTopicName = RegexUtils.getMatchString(topicName, TOPIC_NAME_REGEX);
            String formattedTopicName = tamilTopicName.replaceAll("\\{", "").replaceAll("\\}", "");
            return StringUtils.isNotBlank(formattedTopicName) ? formattedTopicName : topicName;
        }
        return "";
    }

   public String parseEnglishName(String topicName)
    {
        if (StringUtils.isNotBlank(topicName)) {
            return topicName.replaceAll(TOPIC_NAME_REGEX, "");
        }
        return "";
    }
}

File: app/src/main/java/org/worshipsongs/service/DatabaseService.kt
package org.worshipsongs.service

import android.content.Context
import android.database.sqlite.SQLiteDatabase

import org.apache.commons.lang3.StringUtils
import org.worshipsongs.helper.DatabaseHelper
import org.worshipsongs.utils.RegexUtils

import java.io.IOException

/**
 * @Author : Madasamy
 * @Version : 1.0
 */

class DatabaseService
{
    // private var database: SQLiteDatabase? = null
    //var database: SQLiteDatabase? = null
    private var databaseHelper: DatabaseHelper? = null


    val isDatabaseExist: Boolean
        get() = databaseHelper!!.checkDataBase()

    constructor()
    {
        //Do nothing
    }

    constructor(context: Context)
    {
        databaseHelper = DatabaseHelper(context)
    }

    var database: SQLiteDatabase? = null
        get()
        {
            return databaseHelper!!.openDataBase()
        }

    @Throws(IOException::class)
    fun copyDatabase(databasePath: String, dropDatabase: Boolean)
    {
        databaseHelper!!.createDataBase(databasePath, dropDatabase)
    }

    fun open()
    {
        database = databaseHelper!!.openDataBase()
    }

    fun close()
    {
        databaseHelper!!.close()
    }

    fun get(): SQLiteDatabase
    {
        if (database == null)
        {
            database = databaseHelper!!.openDataBase()
        }
        return database!!
    }

    fun parseTamilName(topicName: String?): String
    {
        if (topicName != null && StringUtils.isNotBlank(topicName))
        {
            val tamilTopicName = RegexUtils.getMatchString(topicName, TOPIC_NAME_REGEX)
            val formattedTopicName = tamilTopicName.replace("\\{".toRegex(), "").replace("\\}".toRegex(), "")
            return if (StringUtils.isNotBlank(formattedTopicName)) formattedTopicName else topicName
        }
        return ""
    }

    fun parseEnglishName(topicName: String?): String
    {
        return if (topicName != null && StringUtils.isNotBlank(topicName))
        {
            topicName.replace(TOPIC_NAME_REGEX.toRegex(), "")
        } else ""
    }

    companion object
    {
        val TOPIC_NAME_REGEX = "\\{.*\\}"
    }
}
